packages/react/src/components/Authenticator/FederatedIdentities/context/UseHandleSignInWithRedirectContext.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import {
+  SignInWithRedirectProviderInput,
+  UseHandleSignInWithRedirectOutput,
+} from '../controls';
+import { signInWithRedirect, SignInWithRedirectInput } from 'aws-amplify/auth';
+import { useDataState } from '@aws-amplify/ui-react-core';
+
+export const UseHandleSignInWithRedirectContext = React.createContext<
+  UseHandleSignInWithRedirectOutput | undefined
+>(undefined);
+
+export function UseHandleSignInWithRedirectProvider({
+  handleSignInWithRedirect = signInWithRedirect,
+  children,
+}: SignInWithRedirectProviderInput): JSX.Element {
+  async function _signInWithRedirectAction(
+    _: void | undefined,
+    input: SignInWithRedirectInput
+  ): Promise<void> {
+    await handleSignInWithRedirect(input);
+  }
+
+  return (
+    <UseHandleSignInWithRedirectContext.Provider
+      value={useDataState(_signInWithRedirectAction, undefined)}
```

Comment:
If we're passing the hook outputs here can we move it outside the props definition and just pass the outputs of the hook here?

Reply:
Yep, I can do that. Good catch, would definitely make things clearer.